<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MuData">
<title>CITE-seq data with CiteFuse and MuData • MuData</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="CITE-seq data with CiteFuse and MuData">
<meta property="og:description" content="MuData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MuData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-2">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Blood-CITE-seq.html">CITE-seq data with CiteFuse and MuData</a>
    <a class="dropdown-item" href="../articles/Cord-Blood-CITE-seq.html">CITE-seq data with MultiAssayExperiment and MuData</a>
    <a class="dropdown-item" href="../articles/Getting-Started.html">Getting started with MuData for MultiAssayExperiment</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/PMBio/MuDataMAE/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">
<script src="Blood-CITE-seq_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>CITE-seq data with CiteFuse and MuData</h1>
                        <h4 data-toc-skip class="author">Danila Bredikhin</h4>
            <address class="author_afil">
      European Molecular Biology Laboratory, Heidelberg, Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:danila.bredikhin@embl.de" class="email">danila.bredikhin@embl.de</a>
      </address>
                              <h4 data-toc-skip class="author">Ilia Kats</h4>
            <address class="author_afil">
      German Cancer Research Center, Heidelberg, Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:i.kats@dkfz-heidelberg.de" class="email">i.kats@dkfz-heidelberg.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">2021-12-14</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/PMBio/MuDataMAE/blob/HEAD/vignettes/Blood-CITE-seq.Rmd" class="external-link"><code>vignettes/Blood-CITE-seq.Rmd</code></a></small>
      <div class="d-none name"><code>Blood-CITE-seq.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>CITE-seq data provide RNA and surface protein counts for the same cells. This tutorial shows how MuData can be integrated into with Bioconductor workflows to analyse CITE-seq data.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Stable version of <code>MuData</code> will be available in future bioconductor versions.</p>
<p>BiocManager::install(“MuData”)</p>
<p>The most recent dev build can be installed from GitHub:</p>
<p>library(remotes)remotes::install_github(“PMBio/MuDataMAE”)</p>
</div>
<div class="section level2">
<h2 id="loading-libraries">Loading libraries<a class="anchor" aria-label="anchor" href="#loading-libraries"></a>
</h2>
<p>library(MuData)library(SingleCellExperiment)library(MultiAssayExperiment)library(CiteFuse)library(scater)library(rhdf5)</p>
</div>
<div class="section level2">
<h2 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<p>We will use CITE-seq data available within <a href="http://www.bioconductor.org/packages/release/bioc/html/CiteFuse.html" class="external-link"><code>CiteFuse</code> Bioconductor package</a>.</p>
<p>data(“CITEseq_example”, package = “CiteFuse”)lapply(CITEseq_example, dim)#&gt; $RNA #&gt; [1] 19521 500 #&gt; #&gt; $ADT #&gt; [1] 49 500 #&gt; #&gt; $HTO #&gt; [1] 4 500</p>
<p>This dataset contains three matrices — one with <code>RNA</code> counts, one with antibody-derived tags (<code>ADT</code>) counts and one with hashtag oligonucleotide (<code>HTO</code>) counts.</p>
</div>
<div class="section level2">
<h2 id="processing-count-matrices">Processing count matrices<a class="anchor" aria-label="anchor" href="#processing-count-matrices"></a>
</h2>
<p>While CITE-seq analysis workflows such as <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/CiteFuse/inst/doc/CiteFuse.html" class="external-link">CiteFuse</a> should be consulted for more details, below we exemplify simple data transformations in order to demonstrate how their output can be saved to an H5MU file later on.</p>
<p>Following the CiteFuse tutorial, we start with creating a <code>SingleCellExperiment</code> object with the three matrices:</p>
<p>sce_citeseq &lt;- preprocessing(CITEseq_example)sce_citeseq#&gt; class: SingleCellExperiment #&gt; dim: 19521 500 #&gt; metadata(0): #&gt; assays(1): counts #&gt; rownames(19521): hg19_AL627309.1 hg19_AL669831.5 … hg19_MT-ND6 #&gt; hg19_MT-CYB #&gt; rowData names(0): #&gt; colnames(500): AAGCCGCGTTGTCTTT GATCGCGGTTATCGGT … TTGGCAACACTAGTAC #&gt; GCTGCGAGTTGTGGCC #&gt; colData names(0): #&gt; reducedDimNames(0): #&gt; mainExpName: NULL #&gt; altExpNames(2): ADT HTO</p>
<p>We will add a new assay with normalised RNA counts:</p>
<p>sce_citeseq &lt;- scater::logNormCounts(sce_citeseq)sce_citeseq # new assay: logcounts#&gt; class: SingleCellExperiment #&gt; dim: 19521 500 #&gt; metadata(0): #&gt; assays(2): counts logcounts #&gt; rownames(19521): hg19_AL627309.1 hg19_AL669831.5 … hg19_MT-ND6 #&gt; hg19_MT-CYB #&gt; rowData names(0): #&gt; colnames(500): AAGCCGCGTTGTCTTT GATCGCGGTTATCGGT … TTGGCAACACTAGTAC #&gt; GCTGCGAGTTGTGGCC #&gt; colData names(1): sizeFactor #&gt; reducedDimNames(0): #&gt; mainExpName: NULL #&gt; altExpNames(2): ADT HTO</p>
<p>To the ADT modality, we will add an assay with normalised counts:</p>
<p>sce_citeseq &lt;- CiteFuse::normaliseExprs(sce_citeseq, altExp_name = “ADT”, transform = “log”)altExp(sce_citeseq, “ADT”) # new assay: logcounts#&gt; class: SummarizedExperiment #&gt; dim: 49 500 #&gt; metadata(0): #&gt; assays(2): counts logcounts #&gt; rownames(49): B220 (CD45R) B7-H1 (PD-L1) … TCRb TCRg #&gt; rowData names(0): #&gt; colnames(500): AAGCCGCGTTGTCTTT GATCGCGGTTATCGGT … TTGGCAACACTAGTAC #&gt; GCTGCGAGTTGTGGCC #&gt; colData names(0):</p>
<p>We will also generate reduced dimensions:</p>
<p>sce_citeseq &lt;- scater::runPCA(sce_citeseq, exprs_values = “logcounts”, ncomponents = 20)</p>
<p>scater::plotReducedDim(sce_citeseq, dimred = “PCA”, by_exprs_values = “logcounts”, colour_by = “CD27”)/tmp/Rtmpe9psKV/file2bcb43454773/articles/Blood-CITE-seq_files/figure-html/unnamed-chunk-9-1.png</p>
</div>
<div class="section level2">
<h2 id="making-a-multiassayexperiment-object">Making a MultiAssayExperiment object<a class="anchor" aria-label="anchor" href="#making-a-multiassayexperiment-object"></a>
</h2>
<p>An appropriate structure for multimodal datasets is <a href="https://bioconductor.org/packages/MultiAssayExperiment/" class="external-link"><code>MultiAssayExperiment</code></a>.</p>
<p>We will make a respective MultiAssayExperiment object from <code>sce_citeseq</code>:</p>
<p>experiments &lt;- list( ADT = altExp(sce_citeseq, “ADT”), HTO = altExp(sce_citeseq, “HTO”) )# Drop other modalities from sce_citeseqaltExp(sce_citeseq) &lt;- NULLexperiments[[“RNA”]] &lt;- sce_citeseqmae &lt;- MultiAssayExperiment(experiments)</p>
</div>
<div class="section level2">
<h2 id="writing-to-h5mu">Writing to H5MU<a class="anchor" aria-label="anchor" href="#writing-to-h5mu"></a>
</h2>
<p>We can write the contents of the MultiAssayExperiment object into an H5MU file:</p>
<p>WriteH5MU(mae, “citefuse_example.h5mu”)</p>
<p>We can check that all the modalities were written to the file:</p>
<p>h5 &lt;- rhdf5::H5Fopen(“citefuse_example.h5mu”)h5&amp;‘mod’#&gt; HDF5 GROUP #&gt; name /mod #&gt; filename #&gt; #&gt; name otype dclass dim #&gt; 0 ADT H5I_GROUP<br>
#&gt; 1 HTO H5I_GROUP<br>
#&gt; 2 RNA H5I_GROUP</p>
<p>… both assays for ADT — raw counts are stored in <code>X</code> and normalised counts are in the corresponding layer:</p>
<p>h5&amp;‘mod’&amp;‘ADT’#&gt; HDF5 GROUP #&gt; name /mod/ADT #&gt; filename #&gt; #&gt; name otype dclass dim #&gt; 0 X H5I_GROUP<br>
#&gt; 1 layers H5I_GROUP<br>
#&gt; 2 obs H5I_GROUP<br>
#&gt; 3 var H5I_GROUP h5&amp;‘mod’&amp;‘ADT’&amp;‘layers’#&gt; HDF5 GROUP #&gt; name /mod/ADT/layers #&gt; filename #&gt; #&gt; name otype dclass dim #&gt; 0 logcounts H5I_DATASET FLOAT 49 x 500</p>
<p>… as well as reduced dimensions (PCA):</p>
<p>h5&amp;‘mod’&amp;‘RNA’&amp;‘obsm’#&gt; HDF5 GROUP #&gt; name /mod/RNA/obsm #&gt; filename #&gt; #&gt; name otype dclass dim #&gt; 0 PCA H5I_DATASET FLOAT 20 x 500 rhdf5::H5close()</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li><p><a href="https://www.biorxiv.org/content/10.1101/2021.06.01.445670" class="external-link">Muon: multimodal omics analysis framework</a> preprint</p></li>
<li><p><a href="https://mudata.readthedocs.io/" class="external-link">mudata</a> (Python) documentation</p></li>
<li><p>muon <a href="https://muon.readthedocs.io/" class="external-link">documentation</a> and <a href="https://gtca.github.io/muon/" class="external-link">web page</a></p></li>
<li><p>Kim HJ, Lin Y, Geddes TA, Yang P, Yang JYH (2020). “CiteFuse enables multi-modal analysis of CITE-seq data.” Bioinformatics, 36(14), 4137–4143. <a href="https://doi.org/10.1093/bioinformatics/btaa282" class="external-link uri">https://doi.org/10.1093/bioinformatics/btaa282</a>.</p></li>
<li><p>Ramos M, Schiffer L, Re A, Azhar R, Basunia A, Cabrera CR, Chan T, Chapman P, Davis S, Gomez-Cabrero D, Culhane AC, Haibe-Kains B, Hansen K, Kodali H, Louis MS, Mer AS, Reister M, Morgan M, Carey V, Waldron L (2017). “Software For The Integration Of Multi-Omics Experiments In Bioconductor.” Cancer Research, 77(21); e39-42.</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>sessionInfo()#&gt; R Under development (unstable) (2021-12-11 r81355) #&gt; Platform: x86_64-pc-linux-gnu (64-bit) #&gt; Running under: Ubuntu 20.04.3 LTS #&gt; #&gt; Matrix products: default #&gt; BLAS: /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 #&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0 #&gt; #&gt; locale: #&gt; [1] LC_CTYPE=C.UTF-8 LC_NUMERIC=C LC_TIME=C.UTF-8<br>
#&gt; [4] LC_COLLATE=C.UTF-8 LC_MONETARY=C.UTF-8 LC_MESSAGES=C.UTF-8<br>
#&gt; [7] LC_PAPER=C.UTF-8 LC_NAME=C LC_ADDRESS=C<br>
#&gt; [10] LC_TELEPHONE=C LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C<br>
#&gt; #&gt; attached base packages: #&gt; [1] stats4 stats graphics grDevices utils datasets methods<br>
#&gt; [8] base<br>
#&gt; #&gt; other attached packages: #&gt; [1] rhdf5_2.39.2 scater_1.23.1<br>
#&gt; [3] ggplot2_3.3.5 scuttle_1.5.0<br>
#&gt; [5] CiteFuse_1.7.0 MultiAssayExperiment_1.21.5 #&gt; [7] SingleCellExperiment_1.17.2 SummarizedExperiment_1.25.3 #&gt; [9] Biobase_2.55.0 GenomicRanges_1.47.5<br>
#&gt; [11] GenomeInfoDb_1.31.1 IRanges_2.29.1<br>
#&gt; [13] MatrixGenerics_1.7.0 matrixStats_0.61.0<br>
#&gt; [15] MuData_0.99.0 S4Vectors_0.33.8<br>
#&gt; [17] BiocGenerics_0.41.2 Matrix_1.4-0<br>
#&gt; [19] BiocStyle_2.23.0<br>
#&gt; #&gt; loaded via a namespace (and not attached): #&gt; [1] systemfonts_1.0.3 plyr_1.8.6<br>
#&gt; [3] igraph_1.2.9 splines_4.2.0<br>
#&gt; [5] BiocParallel_1.29.5 digest_0.6.29<br>
#&gt; [7] htmltools_0.5.2 viridis_0.6.2<br>
#&gt; [9] fansi_0.5.0 magrittr_2.0.1<br>
#&gt; [11] memoise_2.0.1 ScaledMatrix_1.3.0<br>
#&gt; [13] cluster_2.1.2 mixtools_1.2.0<br>
#&gt; [15] limma_3.51.1 graphlayouts_0.7.2<br>
#&gt; [17] pkgdown_2.0.1 colorspace_2.0-2<br>
#&gt; [19] ggrepel_0.9.1 textshaping_0.3.6<br>
#&gt; [21] xfun_0.28 dplyr_1.0.7<br>
#&gt; [23] crayon_1.4.2 RCurl_1.98-1.5<br>
#&gt; [25] jsonlite_1.7.2 survival_3.2-13<br>
#&gt; [27] glue_1.5.1 polyclip_1.10-0<br>
#&gt; [29] gtable_0.3.0 zlibbioc_1.41.0<br>
#&gt; [31] XVector_0.35.0 DelayedArray_0.21.2<br>
#&gt; [33] BiocSingular_1.11.0 kernlab_0.9-29<br>
#&gt; [35] Rhdf5lib_1.17.0 scales_1.1.1<br>
#&gt; [37] pheatmap_1.0.12 DBI_1.1.1<br>
#&gt; [39] edgeR_3.37.0 Rcpp_1.0.7<br>
#&gt; [41] viridisLite_0.4.0 dqrng_0.3.0<br>
#&gt; [43] rsvd_1.0.5 propr_4.2.6<br>
#&gt; [45] metapod_1.3.0 RColorBrewer_1.1-2<br>
#&gt; [47] ellipsis_0.3.2 pkgconfig_2.0.3<br>
#&gt; [49] farver_2.1.0 sass_0.4.0<br>
#&gt; [51] uwot_0.1.11 locfit_1.5-9.4<br>
#&gt; [53] utf8_1.2.2 labeling_0.4.2<br>
#&gt; [55] tidyselect_1.1.1 rlang_0.4.12<br>
#&gt; [57] reshape2_1.4.4 munsell_0.5.0<br>
#&gt; [59] tools_4.2.0 cachem_1.0.6<br>
#&gt; [61] dbscan_1.1-8 generics_0.1.1<br>
#&gt; [63] ggridges_0.5.3 evaluate_0.14<br>
#&gt; [65] stringr_1.4.0 fastmap_1.1.0<br>
#&gt; [67] yaml_2.2.1 ragg_1.2.1<br>
#&gt; [69] knitr_1.36 fs_1.5.2<br>
#&gt; [71] tidygraph_1.2.0 purrr_0.3.4<br>
#&gt; [73] randomForest_4.6-14 ggraph_2.0.5<br>
#&gt; [75] sparseMatrixStats_1.7.0 scran_1.23.1<br>
#&gt; [77] compiler_4.2.0 beeswarm_0.4.0<br>
#&gt; [79] tibble_3.1.6 statmod_1.4.36<br>
#&gt; [81] tweenr_1.0.2 bslib_0.3.1<br>
#&gt; [83] stringi_1.7.6 desc_1.4.0<br>
#&gt; [85] lattice_0.20-45 bluster_1.5.0<br>
#&gt; [87] vctrs_0.3.8 pillar_1.6.4<br>
#&gt; [89] lifecycle_1.0.1 rhdf5filters_1.7.0<br>
#&gt; [91] BiocManager_1.30.16 jquerylib_0.1.4<br>
#&gt; [93] BiocNeighbors_1.13.0 cowplot_1.1.1<br>
#&gt; [95] bitops_1.0-7 irlba_2.3.5<br>
#&gt; [97] R6_2.5.1 bookdown_0.24<br>
#&gt; [99] gridExtra_2.3 vipor_0.4.5<br>
#&gt; [101] MASS_7.3-54 assertthat_0.2.1<br>
#&gt; [103] rprojroot_2.0.2 withr_2.4.3<br>
#&gt; [105] GenomeInfoDbData_1.2.7 parallel_4.2.0<br>
#&gt; [107] grid_4.2.0 beachmat_2.11.0<br>
#&gt; [109] tidyr_1.1.4 rmarkdown_2.11<br>
#&gt; [111] DelayedMatrixStats_1.17.0 segmented_1.3-4<br>
#&gt; [113] Rtsne_0.15 ggforce_0.3.3<br>
#&gt; [115] ggbeeswarm_0.6.0</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Danila Bredikhin, Ilia Kats.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
