<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MuData">
<title>CITE-seq data with CiteFuse and MuData • MuData</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="CITE-seq data with CiteFuse and MuData">
<meta property="og:description" content="MuData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MuData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-2">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Blood-CITE-seq.html">CITE-seq data with CiteFuse and MuData</a>
    <a class="dropdown-item" href="../articles/Cord-Blood-CITE-seq.html">CITE-seq data with MultiAssayExperiment and MuData</a>
    <a class="dropdown-item" href="../articles/Getting-Started.html">Getting started with MuData for MultiAssayExperiment</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/PMBio/MuDataMAE/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">
<script src="Blood-CITE-seq_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>CITE-seq data with CiteFuse and MuData</h1>
                        <h4 data-toc-skip class="author">Danila Bredikhin</h4>
            <address class="author_afil">
      European Molecular Biology Laboratory, Heidelberg, Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:danila.bredikhin@embl.de" class="email">danila.bredikhin@embl.de</a>
      </address>
                              <h4 data-toc-skip class="author">Ilia Kats</h4>
            <address class="author_afil">
      German Cancer Research Center, Heidelberg, Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:i.kats@dkfz-heidelberg.de" class="email">i.kats@dkfz-heidelberg.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">2021-12-14</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/PMBio/MuDataMAE/blob/HEAD/vignettes/Blood-CITE-seq.Rmd" class="external-link"><code>vignettes/Blood-CITE-seq.Rmd</code></a></small>
      <div class="d-none name"><code>Blood-CITE-seq.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>CITE-seq data provide RNA and surface protein counts for the same cells. This tutorial shows how MuData can be integrated into with Bioconductor workflows to analyse CITE-seq data.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Stable version of <code>MuData</code> will be available in future bioconductor versions.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"MuData"</span><span class="op">)</span></code></pre></div>
<p>The most recent dev build can be installed from GitHub:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://remotes.r-lib.org" class="external-link">remotes</a></span><span class="op">)</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"PMBio/MuDataMAE"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-libraries">Loading libraries<a class="anchor" aria-label="anchor" href="#loading-libraries"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/PMBio/MuDataMAE" class="external-link">MuData</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/" class="external-link">MultiAssayExperiment</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">CiteFuse</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/grimbough/rhdf5" class="external-link">rhdf5</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<p>We will use CITE-seq data available within <a href="http://www.bioconductor.org/packages/release/bioc/html/CiteFuse.html" class="external-link"><code>CiteFuse</code> Bioconductor package</a>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"CITEseq_example"</span>, package <span class="op">=</span> <span class="st">"CiteFuse"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">CITEseq_example</span>, <span class="va">dim</span><span class="op">)</span>
<span class="co">#&gt; $RNA</span>
<span class="co">#&gt; [1] 19521   500</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $ADT</span>
<span class="co">#&gt; [1]  49 500</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $HTO</span>
<span class="co">#&gt; [1]   4 500</span></code></pre></div>
<p>This dataset contains three matrices — one with <code>RNA</code> counts, one with antibody-derived tags (<code>ADT</code>) counts and one with hashtag oligonucleotide (<code>HTO</code>) counts.</p>
</div>
<div class="section level2">
<h2 id="processing-count-matrices">Processing count matrices<a class="anchor" aria-label="anchor" href="#processing-count-matrices"></a>
</h2>
<p>While CITE-seq analysis workflows such as <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/CiteFuse/inst/doc/CiteFuse.html" class="external-link">CiteFuse</a> should be consulted for more details, below we exemplify simple data transformations in order to demonstrate how their output can be saved to an H5MU file later on.</p>
<p>Following the CiteFuse tutorial, we start with creating a <code>SingleCellExperiment</code> object with the three matrices:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/preprocessing.html" class="external-link">preprocessing</a></span><span class="op">(</span><span class="va">CITEseq_example</span><span class="op">)</span>
<span class="va">sce_citeseq</span>
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 19521 500 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(19521): hg19_AL627309.1 hg19_AL669831.5 ... hg19_MT-ND6</span>
<span class="co">#&gt;   hg19_MT-CYB</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(500): AAGCCGCGTTGTCTTT GATCGCGGTTATCGGT ... TTGGCAACACTAGTAC</span>
<span class="co">#&gt;   GCTGCGAGTTGTGGCC</span>
<span class="co">#&gt; colData names(0):</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; mainExpName: NULL</span>
<span class="co">#&gt; altExpNames(2): ADT HTO</span></code></pre></div>
<p>We will add a new assay with normalised RNA counts:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scuttle/man/logNormCounts.html" class="external-link">logNormCounts</a></span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span>
<span class="va">sce_citeseq</span>  <span class="co"># new assay: logcounts</span>
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 19521 500 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(2): counts logcounts</span>
<span class="co">#&gt; rownames(19521): hg19_AL627309.1 hg19_AL669831.5 ... hg19_MT-ND6</span>
<span class="co">#&gt;   hg19_MT-CYB</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(500): AAGCCGCGTTGTCTTT GATCGCGGTTATCGGT ... TTGGCAACACTAGTAC</span>
<span class="co">#&gt;   GCTGCGAGTTGTGGCC</span>
<span class="co">#&gt; colData names(1): sizeFactor</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; mainExpName: NULL</span>
<span class="co">#&gt; altExpNames(2): ADT HTO</span></code></pre></div>
<p>To the ADT modality, we will add an assay with normalised counts:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu">CiteFuse</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/CiteFuse/man/normaliseExprs.html" class="external-link">normaliseExprs</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, altExp_name <span class="op">=</span> <span class="st">"ADT"</span>, transform <span class="op">=</span> <span class="st">"log"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, <span class="st">"ADT"</span><span class="op">)</span>  <span class="co"># new assay: logcounts</span>
<span class="co">#&gt; class: SummarizedExperiment </span>
<span class="co">#&gt; dim: 49 500 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(2): counts logcounts</span>
<span class="co">#&gt; rownames(49): B220 (CD45R) B7-H1 (PD-L1) ... TCRb TCRg</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(500): AAGCCGCGTTGTCTTT GATCGCGGTTATCGGT ... TTGGCAACACTAGTAC</span>
<span class="co">#&gt;   GCTGCGAGTTGTGGCC</span>
<span class="co">#&gt; colData names(0):</span></code></pre></div>
<p>We will also generate reduced dimensions:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sce_citeseq</span> <span class="op">&lt;-</span> <span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/runPCA.html" class="external-link">runPCA</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, exprs_values <span class="op">=</span> <span class="st">"logcounts"</span>, ncomponents <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">scater</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotReducedDim.html" class="external-link">plotReducedDim</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, dimred <span class="op">=</span> <span class="st">"PCA"</span>, by_exprs_values <span class="op">=</span> <span class="st">"logcounts"</span>, colour_by <span class="op">=</span> <span class="st">"CD27"</span><span class="op">)</span></code></pre></div>
<p><img src="Blood-CITE-seq_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="making-a-multiassayexperiment-object">Making a MultiAssayExperiment object<a class="anchor" aria-label="anchor" href="#making-a-multiassayexperiment-object"></a>
</h2>
<p>An appropriate structure for multimodal datasets is <a href="https://bioconductor.org/packages/MultiAssayExperiment/" class="external-link"><code>MultiAssayExperiment</code></a>.</p>
<p>We will make a respective MultiAssayExperiment object from <code>sce_citeseq</code>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">experiments</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
  ADT <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, <span class="st">"ADT"</span><span class="op">)</span>,
  HTO <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">sce_citeseq</span>, <span class="st">"HTO"</span><span class="op">)</span>
<span class="op">)</span>

<span class="co"># Drop other modalities from sce_citeseq</span>
<span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">sce_citeseq</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span>
<span class="va">experiments</span><span class="op">[[</span><span class="st">"RNA"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="va">sce_citeseq</span>

<span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://waldronlab.io/MultiAssayExperiment/reference/MultiAssayExperiment.html" class="external-link">MultiAssayExperiment</a></span><span class="op">(</span><span class="va">experiments</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="writing-to-h5mu">Writing to H5MU<a class="anchor" aria-label="anchor" href="#writing-to-h5mu"></a>
</h2>
<p>We can write the contents of the MultiAssayExperiment object into an H5MU file:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/WriteH5MU.html">WriteH5MU</a></span><span class="op">(</span><span class="va">mae</span>, <span class="st">"citefuse_example.h5mu"</span><span class="op">)</span></code></pre></div>
<p>We can check that all the modalities were written to the file:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h5</span> <span class="op">&lt;-</span> <span class="fu">rhdf5</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Fopen.html" class="external-link">H5Fopen</a></span><span class="op">(</span><span class="st">"citefuse_example.h5mu"</span><span class="op">)</span>
<span class="va">h5</span><span class="op">&amp;</span><span class="st">'mod'</span>
<span class="co">#&gt; HDF5 GROUP </span>
<span class="co">#&gt;         name /mod</span>
<span class="co">#&gt;     filename </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   name     otype dclass dim</span>
<span class="co">#&gt; 0  ADT H5I_GROUP           </span>
<span class="co">#&gt; 1  HTO H5I_GROUP           </span>
<span class="co">#&gt; 2  RNA H5I_GROUP</span></code></pre></div>
<p>… both assays for ADT — raw counts are stored in <code>X</code> and normalised counts are in the corresponding layer:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h5</span><span class="op">&amp;</span><span class="st">'mod'</span><span class="op">&amp;</span><span class="st">'ADT'</span>
<span class="co">#&gt; HDF5 GROUP </span>
<span class="co">#&gt;         name /mod/ADT</span>
<span class="co">#&gt;     filename </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     name     otype dclass dim</span>
<span class="co">#&gt; 0 X      H5I_GROUP           </span>
<span class="co">#&gt; 1 layers H5I_GROUP           </span>
<span class="co">#&gt; 2 obs    H5I_GROUP           </span>
<span class="co">#&gt; 3 var    H5I_GROUP</span>
<span class="va">h5</span><span class="op">&amp;</span><span class="st">'mod'</span><span class="op">&amp;</span><span class="st">'ADT'</span><span class="op">&amp;</span><span class="st">'layers'</span>
<span class="co">#&gt; HDF5 GROUP </span>
<span class="co">#&gt;         name /mod/ADT/layers</span>
<span class="co">#&gt;     filename </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;        name       otype dclass      dim</span>
<span class="co">#&gt; 0 logcounts H5I_DATASET  FLOAT 49 x 500</span></code></pre></div>
<p>… as well as reduced dimensions (PCA):</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h5</span><span class="op">&amp;</span><span class="st">'mod'</span><span class="op">&amp;</span><span class="st">'RNA'</span><span class="op">&amp;</span><span class="st">'obsm'</span>
<span class="co">#&gt; HDF5 GROUP </span>
<span class="co">#&gt;         name /mod/RNA/obsm</span>
<span class="co">#&gt;     filename </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;   name       otype dclass      dim</span>
<span class="co">#&gt; 0  PCA H5I_DATASET  FLOAT 20 x 500</span>
<span class="fu">rhdf5</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5functions.html" class="external-link">H5close</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li><p><a href="https://www.biorxiv.org/content/10.1101/2021.06.01.445670" class="external-link">Muon: multimodal omics analysis framework</a> preprint</p></li>
<li><p><a href="https://mudata.readthedocs.io/" class="external-link">mudata</a> (Python) documentation</p></li>
<li><p>muon <a href="https://muon.readthedocs.io/" class="external-link">documentation</a> and <a href="https://gtca.github.io/muon/" class="external-link">web page</a></p></li>
<li><p>Kim HJ, Lin Y, Geddes TA, Yang P, Yang JYH (2020). “CiteFuse enables multi-modal analysis of CITE-seq data.” Bioinformatics, 36(14), 4137–4143. <a href="https://doi.org/10.1093/bioinformatics/btaa282" class="external-link uri">https://doi.org/10.1093/bioinformatics/btaa282</a>.</p></li>
<li><p>Ramos M, Schiffer L, Re A, Azhar R, Basunia A, Cabrera CR, Chan T, Chapman P, Davis S, Gomez-Cabrero D, Culhane AC, Haibe-Kains B, Hansen K, Kodali H, Louis MS, Mer AS, Reister M, Morgan M, Carey V, Waldron L (2017). “Software For The Integration Of Multi-Omics Experiments In Bioconductor.” Cancer Research, 77(21); e39-42.</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R Under development (unstable) (2021-12-13 r81374)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0</span>
<span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span>
<span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span>
<span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] rhdf5_2.39.2                scater_1.23.1              </span>
<span class="co">#&gt;  [3] ggplot2_3.3.5               scuttle_1.5.0              </span>
<span class="co">#&gt;  [5] CiteFuse_1.7.0              MultiAssayExperiment_1.21.5</span>
<span class="co">#&gt;  [7] SingleCellExperiment_1.17.2 SummarizedExperiment_1.25.3</span>
<span class="co">#&gt;  [9] Biobase_2.55.0              GenomicRanges_1.47.5       </span>
<span class="co">#&gt; [11] GenomeInfoDb_1.31.1         IRanges_2.29.1             </span>
<span class="co">#&gt; [13] MatrixGenerics_1.7.0        matrixStats_0.61.0         </span>
<span class="co">#&gt; [15] MuData_0.99.0               S4Vectors_0.33.8           </span>
<span class="co">#&gt; [17] BiocGenerics_0.41.2         Matrix_1.4-0               </span>
<span class="co">#&gt; [19] BiocStyle_2.23.0           </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;   [1] systemfonts_1.0.3         plyr_1.8.6               </span>
<span class="co">#&gt;   [3] igraph_1.2.9              splines_4.2.0            </span>
<span class="co">#&gt;   [5] BiocParallel_1.29.5       digest_0.6.29            </span>
<span class="co">#&gt;   [7] htmltools_0.5.2           viridis_0.6.2            </span>
<span class="co">#&gt;   [9] fansi_0.5.0               magrittr_2.0.1           </span>
<span class="co">#&gt;  [11] memoise_2.0.1             ScaledMatrix_1.3.0       </span>
<span class="co">#&gt;  [13] cluster_2.1.2             mixtools_1.2.0           </span>
<span class="co">#&gt;  [15] limma_3.51.1              graphlayouts_0.7.2       </span>
<span class="co">#&gt;  [17] pkgdown_2.0.1             colorspace_2.0-2         </span>
<span class="co">#&gt;  [19] ggrepel_0.9.1             textshaping_0.3.6        </span>
<span class="co">#&gt;  [21] xfun_0.28                 dplyr_1.0.7              </span>
<span class="co">#&gt;  [23] crayon_1.4.2              RCurl_1.98-1.5           </span>
<span class="co">#&gt;  [25] jsonlite_1.7.2            survival_3.2-13          </span>
<span class="co">#&gt;  [27] glue_1.5.1                polyclip_1.10-0          </span>
<span class="co">#&gt;  [29] gtable_0.3.0              zlibbioc_1.41.0          </span>
<span class="co">#&gt;  [31] XVector_0.35.0            DelayedArray_0.21.2      </span>
<span class="co">#&gt;  [33] BiocSingular_1.11.0       kernlab_0.9-29           </span>
<span class="co">#&gt;  [35] Rhdf5lib_1.17.0           scales_1.1.1             </span>
<span class="co">#&gt;  [37] pheatmap_1.0.12           DBI_1.1.1                </span>
<span class="co">#&gt;  [39] edgeR_3.37.0              Rcpp_1.0.7               </span>
<span class="co">#&gt;  [41] viridisLite_0.4.0         dqrng_0.3.0              </span>
<span class="co">#&gt;  [43] rsvd_1.0.5                propr_4.2.6              </span>
<span class="co">#&gt;  [45] metapod_1.3.0             RColorBrewer_1.1-2       </span>
<span class="co">#&gt;  [47] ellipsis_0.3.2            pkgconfig_2.0.3          </span>
<span class="co">#&gt;  [49] farver_2.1.0              sass_0.4.0               </span>
<span class="co">#&gt;  [51] uwot_0.1.11               locfit_1.5-9.4           </span>
<span class="co">#&gt;  [53] utf8_1.2.2                labeling_0.4.2           </span>
<span class="co">#&gt;  [55] tidyselect_1.1.1          rlang_0.4.12             </span>
<span class="co">#&gt;  [57] reshape2_1.4.4            munsell_0.5.0            </span>
<span class="co">#&gt;  [59] tools_4.2.0               cachem_1.0.6             </span>
<span class="co">#&gt;  [61] dbscan_1.1-8              generics_0.1.1           </span>
<span class="co">#&gt;  [63] ggridges_0.5.3            evaluate_0.14            </span>
<span class="co">#&gt;  [65] stringr_1.4.0             fastmap_1.1.0            </span>
<span class="co">#&gt;  [67] yaml_2.2.1                ragg_1.2.1               </span>
<span class="co">#&gt;  [69] knitr_1.36                fs_1.5.2                 </span>
<span class="co">#&gt;  [71] tidygraph_1.2.0           purrr_0.3.4              </span>
<span class="co">#&gt;  [73] randomForest_4.6-14       ggraph_2.0.5             </span>
<span class="co">#&gt;  [75] sparseMatrixStats_1.7.0   scran_1.23.1             </span>
<span class="co">#&gt;  [77] compiler_4.2.0            beeswarm_0.4.0           </span>
<span class="co">#&gt;  [79] tibble_3.1.6              statmod_1.4.36           </span>
<span class="co">#&gt;  [81] tweenr_1.0.2              bslib_0.3.1              </span>
<span class="co">#&gt;  [83] stringi_1.7.6             highr_0.9                </span>
<span class="co">#&gt;  [85] desc_1.4.0                lattice_0.20-45          </span>
<span class="co">#&gt;  [87] bluster_1.5.0             vctrs_0.3.8              </span>
<span class="co">#&gt;  [89] pillar_1.6.4              lifecycle_1.0.1          </span>
<span class="co">#&gt;  [91] rhdf5filters_1.7.0        BiocManager_1.30.16      </span>
<span class="co">#&gt;  [93] jquerylib_0.1.4           BiocNeighbors_1.13.0     </span>
<span class="co">#&gt;  [95] cowplot_1.1.1             bitops_1.0-7             </span>
<span class="co">#&gt;  [97] irlba_2.3.5               R6_2.5.1                 </span>
<span class="co">#&gt;  [99] bookdown_0.24             gridExtra_2.3            </span>
<span class="co">#&gt; [101] vipor_0.4.5               MASS_7.3-54              </span>
<span class="co">#&gt; [103] assertthat_0.2.1          rprojroot_2.0.2          </span>
<span class="co">#&gt; [105] withr_2.4.3               GenomeInfoDbData_1.2.7   </span>
<span class="co">#&gt; [107] parallel_4.2.0            grid_4.2.0               </span>
<span class="co">#&gt; [109] beachmat_2.11.0           tidyr_1.1.4              </span>
<span class="co">#&gt; [111] rmarkdown_2.11            DelayedMatrixStats_1.17.0</span>
<span class="co">#&gt; [113] segmented_1.3-4           Rtsne_0.15               </span>
<span class="co">#&gt; [115] ggforce_0.3.3             ggbeeswarm_0.6.0</span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Danila Bredikhin, Ilia Kats.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
