<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MuData">
<title>CITE-seq data with MultiAssayExperiment and MuData • MuData</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="CITE-seq data with MultiAssayExperiment and MuData">
<meta property="og:description" content="MuData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MuData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-2">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Blood-CITE-seq.html">CITE-seq data with CiteFuse and MuData</a>
    <a class="dropdown-item" href="../articles/Cord-Blood-CITE-seq.html">CITE-seq data with MultiAssayExperiment and MuData</a>
    <a class="dropdown-item" href="../articles/Getting-Started.html">Getting started with MuData for MultiAssayExperiment</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/PMBio/MuDataMAE/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">
<script src="Cord-Blood-CITE-seq_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>CITE-seq data with MultiAssayExperiment and MuData</h1>
                        <h4 data-toc-skip class="author">Danila Bredikhin</h4>
            <address class="author_afil">
      European Molecular Biology Laboratory, Heidelberg, Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:danila.bredikhin@embl.de" class="email">danila.bredikhin@embl.de</a>
      </address>
                              <h4 data-toc-skip class="author">Ilia Kats</h4>
            <address class="author_afil">
      German Cancer Research Center, Heidelberg, Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:i.kats@dkfz-heidelberg.de" class="email">i.kats@dkfz-heidelberg.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">2021-12-13</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/PMBio/MuDataMAE/blob/HEAD/vignettes/Cord-Blood-CITE-seq.Rmd" class="external-link"><code>vignettes/Cord-Blood-CITE-seq.Rmd</code></a></small>
      <div class="d-none name"><code>Cord-Blood-CITE-seq.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>CITE-seq data provide RNA and surface protein counts for the same cells. This tutorial shows how MuData can be integrated into with Bioconductor workflows to analyse CITE-seq data.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Stable version of <code>MuData</code> will be available in future bioconductor versions.</p>
<p>BiocManager::install(“MuData”)</p>
<p>The most recent dev build can be installed from GitHub:</p>
<p>library(remotes)remotes::install_github(“PMBio/MuDataMAE”)</p>
</div>
<div class="section level2">
<h2 id="loading-libraries">Loading libraries<a class="anchor" aria-label="anchor" href="#loading-libraries"></a>
</h2>
<p>library(MuData)library(SingleCellExperiment)library(MultiAssayExperiment)library(SingleCellMultiModal)library(scater)library(rhdf5)</p>
</div>
<div class="section level2">
<h2 id="loading-data">Loading data<a class="anchor" aria-label="anchor" href="#loading-data"></a>
</h2>
<p>We will use CITE-seq data accessible with the <a href="https://bioconductor.org/packages/release/data/experiment/vignettes/SingleCellMultiModal/inst/doc/CITEseq.html" class="external-link"><code>SingleCellMultiModal</code> Bioconductor package</a>, which was originally described in <a href="https://www.nature.com/articles/nmeth.4380" class="external-link">Stoeckius et al., 2017</a>.</p>
<p>mae &lt;- CITEseq( DataType=“cord_blood”, modes="<em>“, dry.run=FALSE, version=”1.0.0" )#&gt; Dataset: cord_blood #&gt; snapshotDate(): 2021-11-24 #&gt; Working on: scADT_Counts #&gt; Working on: scRNAseq_Counts #&gt; see ?SingleCellMultiModal and browseVignettes(‘SingleCellMultiModal’) for documentation #&gt; downloading 1 resources #&gt; retrieving 1 resource #&gt; loading from cache #&gt; see ?SingleCellMultiModal and browseVignettes(‘SingleCellMultiModal’) for documentation #&gt; downloading 1 resources #&gt; retrieving 1 resource #&gt; loading from cache mae#&gt; A MultiAssayExperiment object of 2 listed #&gt; experiments with user-defined names and respective classes. #&gt; Containing an ExperimentList class object of length 2: #&gt; [1] scADT: matrix with 13 rows and 8617 columns #&gt; [2] scRNAseq: matrix with 36280 rows and 8617 columns #&gt; Functionality: #&gt; experiments() - obtain the ExperimentList instance #&gt; colData() - the primary/phenotype DataFrame #&gt; sampleMap() - the sample coordination DataFrame #&gt; <code>$</code>, <code>[</code>, <code>[[</code> - extract colData columns, subset, or experiment #&gt; </em>Format() - convert into a long or wide DataFrame #&gt; assays() - convert ExperimentList to a SimpleList of matrices #&gt; exportClass() - save data to flat files</p>
<p>We see two modalities in the object — <code>scRNAseq</code> and <code>scADT</code>, the latter providing counts for antibody-derived tags. Notably, each experiment is a matrix.</p>
</div>
<div class="section level2">
<h2 id="processing-adt-data">Processing ADT data<a class="anchor" aria-label="anchor" href="#processing-adt-data"></a>
</h2>
<p>While CITE-seq analysis workflows such as <a href="http://www.bioconductor.org/packages/release/bioc/html/CiteFuse.html" class="external-link">CiteFuse</a> should be consulted for more details, below we exemplify simple data transformation in order to demonstrate how their output can be saved to an H5MU file later on.</p>
<p>For ADT counts, we will apply CLR transformation following <a href="https://doi.org/10.1016/j.cell.2021.04.048" class="external-link">Hao et al., 2020</a>:</p>
</div>
<div class="section level1">
<h1 id="define-clr-transformation-as-in-the-seurat-workflowclr---functiondata-tapplydata-1-functionx-log1px-expsumlog1pxx-0-na-rm-true-lengthx">Define CLR transformation as in the Seurat workflowclr &lt;- function(data) t(apply(data, 1, function(x) log1p(x / (exp(sum(log1p(x[x &gt; 0]), na.rm = TRUE) / length(x))))))<a class="anchor" aria-label="anchor" href="#define-clr-transformation-as-in-the-seurat-workflowclr---functiondata-tapplydata-1-functionx-log1px-expsumlog1pxx-0-na-rm-true-lengthx"></a>
</h1>
<p>We will make the ADT modality a <code>SingleCellExperiment</code> object and add an assay with CLR-transformed counts:</p>
<p>adt_counts &lt;- mae[[“scADT”]]mae[[“scADT”]] &lt;- SingleCellExperiment(adt_counts)assays(mae[[“scADT”]])[[“clr”]] &lt;- clr(adt_counts)</p>
<p>We will also generate reduced dimensions taking advantage of the functionality in the <a href="https://bioconductor.org/packages/release/bioc/vignettes/scater/inst/doc/overview.html" class="external-link"><code>scater</code> package</a>:</p>
<p>mae[[“scADT”]] &lt;- runPCA(mae[[“scADT”]], exprs_values = “clr”, ncomponents = 20)#&gt; Warning in check_numbers(k = k, nu = nu, nv = nv, limit = min(dim(x)) - : more #&gt; singular values/vectors requested than available #&gt; Warning in (function (A, nv = 5, nu = nv, maxit = 1000, work = nv + 7, reorth = #&gt; TRUE, : You’re computing too large a percentage of total singular values, use a #&gt; standard svd instead.</p>
<p>plotReducedDim(mae[[“scADT”]], dimred = “PCA”, by_exprs_values = “clr”, colour_by = “CD3”)/tmp/Rtmpx8BRQ3/file144e1165382b6/articles/Cord-Blood-CITE-seq_files/figure-html/unnamed-chunk-8-1.pngplotReducedDim(mae[[“scADT”]], dimred = “PCA”, by_exprs_values = “clr”, colour_by = “CD14”)/tmp/Rtmpx8BRQ3/file144e1165382b6/articles/Cord-Blood-CITE-seq_files/figure-html/unnamed-chunk-8-2.png</p>
<div class="section level2">
<h2 id="writing-h5mu-files">Writing H5MU files<a class="anchor" aria-label="anchor" href="#writing-h5mu-files"></a>
</h2>
<p>We can write the contents of the MultiAssayExperiment object into an H5MU file:</p>
<p>WriteH5MU(mae, “cord_blood_citeseq.h5mu”)</p>
<p>We can check that both modalities were written to the file, whether it was a <code>matrix</code> for RNA or <code>SingleCellExperiment</code> for ADT:</p>
<p>h5 &lt;- rhdf5::H5Fopen(“cord_blood_citeseq.h5mu”)h5&amp;‘mod’#&gt; HDF5 GROUP #&gt; name /mod #&gt; filename #&gt; #&gt; name otype dclass dim #&gt; 0 scADT H5I_GROUP<br>
#&gt; 1 scRNAseq H5I_GROUP</p>
<p>… both assays for ADT — raw counts are stored in <code>X</code> and CLR-transformed counts are in the corresponding layer:</p>
<p>h5&amp;‘mod’&amp;‘scADT’#&gt; HDF5 GROUP #&gt; name /mod/scADT #&gt; filename #&gt; #&gt; name otype dclass dim #&gt; 0 X H5I_DATASET INTEGER 13 x 8617 #&gt; 1 layers H5I_GROUP<br>
#&gt; 2 obs H5I_GROUP<br>
#&gt; 3 obsm H5I_GROUP<br>
#&gt; 4 var H5I_GROUP h5&amp;‘mod’&amp;‘scADT’&amp;‘layers’#&gt; HDF5 GROUP #&gt; name /mod/scADT/layers #&gt; filename #&gt; #&gt; name otype dclass dim #&gt; 0 clr H5I_DATASET FLOAT 13 x 8617</p>
<p>… as well as reduced dimensions (PCA):</p>
<p>h5&amp;‘mod’&amp;‘scADT’&amp;‘obsm’#&gt; HDF5 GROUP #&gt; name /mod/scADT/obsm #&gt; filename #&gt; #&gt; name otype dclass dim #&gt; 0 PCA H5I_DATASET FLOAT 12 x 8617 rhdf5::H5close()</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li><p><a href="https://www.biorxiv.org/content/10.1101/2021.06.01.445670" class="external-link">Muon: multimodal omics analysis framework</a> preprint</p></li>
<li><p><a href="https://mudata.readthedocs.io/" class="external-link">mudata</a> (Python) documentation</p></li>
<li><p>muon <a href="https://muon.readthedocs.io/" class="external-link">documentation</a> and <a href="https://gtca.github.io/muon/" class="external-link">web page</a></p></li>
<li><p>Stoeckius, M., Hafemeister, C., Stephenson, W., Houck-Loomis, B., Chattopadhyay, P.K., Swerdlow, H., Satija, R. and Smibert, P., 2017. Simultaneous epitope and transcriptome measurement in single cells. <em>Nature methods</em>, 14(9), pp.865-868.</p></li>
<li><p>Hao, Y., Hao, S., Andersen-Nissen, E., Mauck III, W.M., Zheng, S., Butler, A., Lee, M.J., Wilk, A.J., Darby, C., Zager, M. and Hoffman, P., 2021. Integrated analysis of multimodal single-cell data. <em>Cell</em>.</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>sessionInfo()#&gt; R Under development (unstable) (2021-12-11 r81355) #&gt; Platform: x86_64-pc-linux-gnu (64-bit) #&gt; Running under: Ubuntu 20.04.3 LTS #&gt; #&gt; Matrix products: default #&gt; BLAS: /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 #&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0 #&gt; #&gt; locale: #&gt; [1] LC_CTYPE=C.UTF-8 LC_NUMERIC=C LC_TIME=C.UTF-8<br>
#&gt; [4] LC_COLLATE=C.UTF-8 LC_MONETARY=C.UTF-8 LC_MESSAGES=C.UTF-8<br>
#&gt; [7] LC_PAPER=C.UTF-8 LC_NAME=C LC_ADDRESS=C<br>
#&gt; [10] LC_TELEPHONE=C LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C<br>
#&gt; #&gt; attached base packages: #&gt; [1] stats4 stats graphics grDevices utils datasets methods<br>
#&gt; [8] base<br>
#&gt; #&gt; other attached packages: #&gt; [1] rhdf5_2.39.2 scater_1.23.1<br>
#&gt; [3] ggplot2_3.3.5 scuttle_1.5.0<br>
#&gt; [5] SingleCellMultiModal_1.7.0 MultiAssayExperiment_1.21.5 #&gt; [7] SingleCellExperiment_1.17.2 SummarizedExperiment_1.25.3 #&gt; [9] Biobase_2.55.0 GenomicRanges_1.47.5<br>
#&gt; [11] GenomeInfoDb_1.31.1 IRanges_2.29.1<br>
#&gt; [13] MatrixGenerics_1.7.0 matrixStats_0.61.0<br>
#&gt; [15] MuData_0.99.0 S4Vectors_0.33.8<br>
#&gt; [17] BiocGenerics_0.41.2 Matrix_1.4-0<br>
#&gt; [19] BiocStyle_2.23.0<br>
#&gt; #&gt; loaded via a namespace (and not attached): #&gt; [1] AnnotationHub_3.3.7 BiocFileCache_2.3.0<br>
#&gt; [3] systemfonts_1.0.3 BiocParallel_1.29.5<br>
#&gt; [5] digest_0.6.29 htmltools_0.5.2<br>
#&gt; [7] viridis_0.6.2 magick_2.7.3<br>
#&gt; [9] fansi_0.5.0 magrittr_2.0.1<br>
#&gt; [11] memoise_2.0.1 ScaledMatrix_1.3.0<br>
#&gt; [13] SpatialExperiment_1.5.0 limma_3.51.1<br>
#&gt; [15] Biostrings_2.63.0 R.utils_2.11.0<br>
#&gt; [17] pkgdown_2.0.1 colorspace_2.0-2<br>
#&gt; [19] blob_1.2.2 rappdirs_0.3.3<br>
#&gt; [21] ggrepel_0.9.1 textshaping_0.3.6<br>
#&gt; [23] xfun_0.28 dplyr_1.0.7<br>
#&gt; [25] crayon_1.4.2 RCurl_1.98-1.5<br>
#&gt; [27] jsonlite_1.7.2 glue_1.5.1<br>
#&gt; [29] gtable_0.3.0 zlibbioc_1.41.0<br>
#&gt; [31] XVector_0.35.0 DelayedArray_0.21.2<br>
#&gt; [33] BiocSingular_1.11.0 DropletUtils_1.15.2<br>
#&gt; [35] Rhdf5lib_1.17.0 HDF5Array_1.23.2<br>
#&gt; [37] scales_1.1.1 DBI_1.1.1<br>
#&gt; [39] edgeR_3.37.0 Rcpp_1.0.7<br>
#&gt; [41] viridisLite_0.4.0 xtable_1.8-4<br>
#&gt; [43] dqrng_0.3.0 bit_4.0.4<br>
#&gt; [45] rsvd_1.0.5 httr_1.4.2<br>
#&gt; [47] ellipsis_0.3.2 pkgconfig_2.0.3<br>
#&gt; [49] R.methodsS3_1.8.1 farver_2.1.0<br>
#&gt; [51] sass_0.4.0 dbplyr_2.1.1<br>
#&gt; [53] locfit_1.5-9.4 utf8_1.2.2<br>
#&gt; [55] tidyselect_1.1.1 labeling_0.4.2<br>
#&gt; [57] rlang_0.4.12 later_1.3.0<br>
#&gt; [59] AnnotationDbi_1.57.1 munsell_0.5.0<br>
#&gt; [61] BiocVersion_3.15.0 tools_4.2.0<br>
#&gt; [63] cachem_1.0.6 generics_0.1.1<br>
#&gt; [65] RSQLite_2.2.9 ExperimentHub_2.3.4<br>
#&gt; [67] evaluate_0.14 stringr_1.4.0<br>
#&gt; [69] fastmap_1.1.0 yaml_2.2.1<br>
#&gt; [71] ragg_1.2.1 knitr_1.36<br>
#&gt; [73] bit64_4.0.5 fs_1.5.2<br>
#&gt; [75] purrr_0.3.4 KEGGREST_1.35.0<br>
#&gt; [77] sparseMatrixStats_1.7.0 mime_0.12<br>
#&gt; [79] R.oo_1.24.0 compiler_4.2.0<br>
#&gt; [81] beeswarm_0.4.0 filelock_1.0.2<br>
#&gt; [83] curl_4.3.2 png_0.1-7<br>
#&gt; [85] interactiveDisplayBase_1.33.0 tibble_3.1.6<br>
#&gt; [87] bslib_0.3.1 stringi_1.7.6<br>
#&gt; [89] desc_1.4.0 lattice_0.20-45<br>
#&gt; [91] vctrs_0.3.8 pillar_1.6.4<br>
#&gt; [93] lifecycle_1.0.1 rhdf5filters_1.7.0<br>
#&gt; [95] BiocManager_1.30.16 jquerylib_0.1.4<br>
#&gt; [97] BiocNeighbors_1.13.0 cowplot_1.1.1<br>
#&gt; [99] bitops_1.0-7 irlba_2.3.5<br>
#&gt; [101] httpuv_1.6.3 R6_2.5.1<br>
#&gt; [103] bookdown_0.24 promises_1.2.0.1<br>
#&gt; [105] gridExtra_2.3 vipor_0.4.5<br>
#&gt; [107] assertthat_0.2.1 rprojroot_2.0.2<br>
#&gt; [109] rjson_0.2.20 withr_2.4.3<br>
#&gt; [111] GenomeInfoDbData_1.2.7 parallel_4.2.0<br>
#&gt; [113] grid_4.2.0 beachmat_2.11.0<br>
#&gt; [115] rmarkdown_2.11 DelayedMatrixStats_1.17.0<br>
#&gt; [117] shiny_1.7.1 ggbeeswarm_0.6.0</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Danila Bredikhin, Ilia Kats.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
