<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting started with MuData for MultiAssayExperiment • MuData</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting started with MuData for MultiAssayExperiment">
<meta property="og:description" content="MuData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MuData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Getting-Started.html">Getting started with MuData for MultiAssayExperiment</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Getting-Started_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting started with MuData for MultiAssayExperiment</h1>
                        <h4 class="author">Danila Bredikhin</h4>
            <address class="author_afil">
      European Molecular Biology Laboratory, Heidelberg, Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:danila.bredikhin@embl.de" class="email">danila.bredikhin@embl.de</a>
      </address>
                              <h4 class="author">Ilia Kats</h4>
            <address class="author_afil">
      German Cancer Research Center, Heidelberg, Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:i.kats@dkfz-heidelberg.de" class="email">i.kats@dkfz-heidelberg.de</a>
      </address>
                  
            <h4 class="date">2021-11-30</h4>
      
      
      <div class="hidden name"><code>Getting-Started.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Multimodal data format — <a href="https://mudata.readthedocs.io/">MuData</a> — <a href="https://www.biorxiv.org/content/10.1101/2021.06.01.445670v1">has been introduced</a> to address the need for cross-platform standard for sharing large-scale multimodal omics data. Importantly, it develops ideas of and is compatible with <a href="https://anndata.readthedocs.io/">AnnData</a> standard for storing raw and derived data for unimodal datasets.</p>
<p>In Bioconductor, multimodal datasets have been stored in <a href="https://bioconductor.org/packages/MultiAssayExperiment">MultiAssayExperiment</a> (MAE) objects. This <code>MuData</code> package provides functionality to read data from MuData files into MAE objects as well as to save MAE objects into H5MU files.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>Stable version of <code>MuData</code> will be available in future bioconductor versions.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="st">"MuData"</span><span class="op">)</span></code></pre></div>
<p>The most recent dev build can be installed from GitHub:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://remotes.r-lib.org">remotes</a></span><span class="op">)</span>
<span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html">install_github</a></span><span class="op">(</span><span class="st">"PMBio/MuDataMAE"</span><span class="op">)</span></code></pre></div>
</div>
<div id="loading-libraries" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-libraries" class="anchor"></a>Loading libraries</h1>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">MuData</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/">MultiAssayExperiment</a></span><span class="op">)</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/grimbough/rhdf5">rhdf5</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="writing-h5mu-files" class="section level1">
<h1 class="hasAnchor">
<a href="#writing-h5mu-files" class="anchor"></a>Writing H5MU files</h1>
<p>We’ll use a simple MAE object from the <code>MultiAssayExperiment</code> package that we’ll then save in a H5MU file.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">miniACC</span><span class="op">)</span>
<span class="va">miniACC</span>
<span class="co">#&gt; A MultiAssayExperiment object of 5 listed</span>
<span class="co">#&gt;  experiments with user-defined names and respective classes.</span>
<span class="co">#&gt;  Containing an ExperimentList class object of length 5:</span>
<span class="co">#&gt;  [1] RNASeq2GeneNorm: SummarizedExperiment with 198 rows and 79 columns</span>
<span class="co">#&gt;  [2] gistict: SummarizedExperiment with 198 rows and 90 columns</span>
<span class="co">#&gt;  [3] RPPAArray: SummarizedExperiment with 33 rows and 46 columns</span>
<span class="co">#&gt;  [4] Mutations: matrix with 97 rows and 90 columns</span>
<span class="co">#&gt;  [5] miRNASeqGene: SummarizedExperiment with 471 rows and 80 columns</span>
<span class="co">#&gt; Functionality:</span>
<span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span>
<span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span>
<span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span>
<span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span>
<span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="co">#&gt;  exportClass() - save data to flat files</span></code></pre></div>
<p>We will now write its contents into an H5MU file with <code>WriteH5MU</code>:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/WriteH5MU.html">WriteH5MU</a></span><span class="op">(</span><span class="va">miniACC</span>, <span class="st">"miniacc.h5mu"</span><span class="op">)</span></code></pre></div>
</div>
<div id="reading-h5mu-files" class="section level1">
<h1 class="hasAnchor">
<a href="#reading-h5mu-files" class="anchor"></a>Reading H5MU files</h1>
<p>We can manually check the top level of the structure of the file:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">rhdf5</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/h5ls.html">h5ls</a></span><span class="op">(</span><span class="st">"miniacc.h5mu"</span>, recursive <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="co">#&gt;   group   name     otype dclass dim</span>
<span class="co">#&gt; 0     /    mod H5I_GROUP           </span>
<span class="co">#&gt; 1     /    obs H5I_GROUP           </span>
<span class="co">#&gt; 2     /   obsm H5I_GROUP           </span>
<span class="co">#&gt; 3     / obsmap H5I_GROUP           </span>
<span class="co">#&gt; 4     /    var H5I_GROUP</span></code></pre></div>
<p>Or dig deeper into the file:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">h5</span> <span class="op">&lt;-</span> <span class="fu">rhdf5</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5Fopen.html">H5Fopen</a></span><span class="op">(</span><span class="st">"miniacc.h5mu"</span><span class="op">)</span>
<span class="va">h5</span><span class="op">&amp;</span><span class="st">'mod'</span>
<span class="co">#&gt; HDF5 GROUP </span>
<span class="co">#&gt;         name /mod</span>
<span class="co">#&gt;     filename </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;              name     otype dclass dim</span>
<span class="co">#&gt; 0 Mutations       H5I_GROUP           </span>
<span class="co">#&gt; 1 RNASeq2GeneNorm H5I_GROUP           </span>
<span class="co">#&gt; 2 RPPAArray       H5I_GROUP           </span>
<span class="co">#&gt; 3 gistict         H5I_GROUP           </span>
<span class="co">#&gt; 4 miRNASeqGene    H5I_GROUP</span>
<span class="fu">rhdf5</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/rhdf5/man/H5functions.html">H5close</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div id="creating-mae-objects-from-h5mu-files" class="section level2">
<h2 class="hasAnchor">
<a href="#creating-mae-objects-from-h5mu-files" class="anchor"></a>Creating MAE objects from H5MU files</h2>
<p>This package provides <code>ReadH5MU</code> to create an object with data from an H5MU file. Since H5MU structure has been designed to accommodate more structured information than MAE, only some data will be read. For instance, MAE has no support for loading multimodal embeddings or pairwise graphs.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">acc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ReadH5MU.html">ReadH5MU</a></span><span class="op">(</span><span class="st">"miniacc.h5mu"</span><span class="op">)</span>
<span class="va">acc</span>
<span class="co">#&gt; A MultiAssayExperiment object of 5 listed</span>
<span class="co">#&gt;  experiments with user-defined names and respective classes.</span>
<span class="co">#&gt;  Containing an ExperimentList class object of length 5:</span>
<span class="co">#&gt;  [1] Mutations: SingleCellExperiment with 97 rows and 90 columns</span>
<span class="co">#&gt;  [2] RNASeq2GeneNorm: SingleCellExperiment with 198 rows and 79 columns</span>
<span class="co">#&gt;  [3] RPPAArray: SingleCellExperiment with 33 rows and 46 columns</span>
<span class="co">#&gt;  [4] gistict: SingleCellExperiment with 198 rows and 90 columns</span>
<span class="co">#&gt;  [5] miRNASeqGene: SingleCellExperiment with 471 rows and 80 columns</span>
<span class="co">#&gt; Functionality:</span>
<span class="co">#&gt;  experiments() - obtain the ExperimentList instance</span>
<span class="co">#&gt;  colData() - the primary/phenotype DataFrame</span>
<span class="co">#&gt;  sampleMap() - the sample coordination DataFrame</span>
<span class="co">#&gt;  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span>
<span class="co">#&gt;  *Format() - convert into a long or wide DataFrame</span>
<span class="co">#&gt;  assays() - convert ExperimentList to a SimpleList of matrices</span>
<span class="co">#&gt;  exportClass() - save data to flat files</span></code></pre></div>
<p>Importantly, we recover the information from the original MAE object:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">miniACC</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 6 rows and 4 columns</span>
<span class="co">#&gt;                 patientID years_to_birth vital_status days_to_death</span>
<span class="co">#&gt;               &lt;character&gt;      &lt;integer&gt;    &lt;integer&gt;     &lt;integer&gt;</span>
<span class="co">#&gt; TCGA-OR-A5J1 TCGA-OR-A5J1             58            1          1355</span>
<span class="co">#&gt; TCGA-OR-A5J2 TCGA-OR-A5J2             44            1          1677</span>
<span class="co">#&gt; TCGA-OR-A5J3 TCGA-OR-A5J3             23            0            NA</span>
<span class="co">#&gt; TCGA-OR-A5J4 TCGA-OR-A5J4             23            1           423</span>
<span class="co">#&gt; TCGA-OR-A5J5 TCGA-OR-A5J5             30            1           365</span>
<span class="co">#&gt; TCGA-OR-A5J6 TCGA-OR-A5J6             29            0            NA</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">colData</a></span><span class="op">(</span><span class="va">acc</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 6 rows and 4 columns</span>
<span class="co">#&gt;                 patientID years_to_birth vital_status days_to_death</span>
<span class="co">#&gt;               &lt;character&gt;      &lt;integer&gt;    &lt;integer&gt;     &lt;integer&gt;</span>
<span class="co">#&gt; TCGA-OR-A5J1 TCGA-OR-A5J1             58            1          1355</span>
<span class="co">#&gt; TCGA-OR-A5J2 TCGA-OR-A5J2             44            1          1677</span>
<span class="co">#&gt; TCGA-OR-A5J3 TCGA-OR-A5J3             23            0            NA</span>
<span class="co">#&gt; TCGA-OR-A5J4 TCGA-OR-A5J4             23            1           423</span>
<span class="co">#&gt; TCGA-OR-A5J5 TCGA-OR-A5J5             30            1           365</span>
<span class="co">#&gt; TCGA-OR-A5J6 TCGA-OR-A5J6             29            0            NA</span></code></pre></div>
<p>Features metadata is also recovered:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">miniACC</span><span class="op">[[</span><span class="st">"gistict"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 6 rows and 3 columns</span>
<span class="co">#&gt;        Gene.Symbol    Locus.ID    Cytoband</span>
<span class="co">#&gt;        &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt; DIRAS3      DIRAS3        9077      1p31.3</span>
<span class="co">#&gt; MAPK14      MAPK14        1432     6p21.31</span>
<span class="co">#&gt; YAP1          YAP1       10413     11q22.1</span>
<span class="co">#&gt; CDKN1B      CDKN1B        1027     12p13.1</span>
<span class="co">#&gt; ERBB2        ERBB2        2064       17q12</span>
<span class="co">#&gt; G6PD          G6PD        2539        Xq28</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">rowData</a></span><span class="op">(</span><span class="va">acc</span><span class="op">[[</span><span class="st">"gistict"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; DataFrame with 6 rows and 3 columns</span>
<span class="co">#&gt;        Gene.Symbol    Locus.ID    Cytoband</span>
<span class="co">#&gt;        &lt;character&gt; &lt;character&gt; &lt;character&gt;</span>
<span class="co">#&gt; DIRAS3      DIRAS3        9077      1p31.3</span>
<span class="co">#&gt; MAPK14      MAPK14        1432     6p21.31</span>
<span class="co">#&gt; YAP1          YAP1       10413     11q22.1</span>
<span class="co">#&gt; CDKN1B      CDKN1B        1027     12p13.1</span>
<span class="co">#&gt; ERBB2        ERBB2        2064       17q12</span>
<span class="co">#&gt; G6PD          G6PD        2539        Xq28</span></code></pre></div>
<div id="backed-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#backed-objects" class="anchor"></a>Backed objects</h3>
<p>It is possible to read H5MU files while keeping matrices (both <code>.X</code> and <code>.layers</code>) on disk.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">acc_b</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ReadH5MU.html">ReadH5MU</a></span><span class="op">(</span><span class="st">"miniacc.h5mu"</span>, backed <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">acc_b</span><span class="op">[[</span><span class="st">"RNASeq2GeneNorm"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>
<span class="co">#&gt; &lt;5 x 3&gt; matrix of class DelayedMatrix and type "double":</span>
<span class="co">#&gt;        TCGA-OR-A5J1-01A-11R-A29S-07 TCGA-OR-A5J2-01A-11R-A29S-07</span>
<span class="co">#&gt; DIRAS3                    1487.0317                       9.6631</span>
<span class="co">#&gt; MAPK14                     778.5783                    2823.6469</span>
<span class="co">#&gt; YAP1                      1009.6061                    2305.0590</span>
<span class="co">#&gt; CDKN1B                    2101.3449                    4248.9584</span>
<span class="co">#&gt; ERBB2                      651.2968                     246.4098</span>
<span class="co">#&gt;        TCGA-OR-A5J3-01A-11R-A29S-07</span>
<span class="co">#&gt; DIRAS3                      18.9602</span>
<span class="co">#&gt; MAPK14                    1061.7686</span>
<span class="co">#&gt; YAP1                      1561.2502</span>
<span class="co">#&gt; CDKN1B                    1348.5410</span>
<span class="co">#&gt; ERBB2                       90.0607</span></code></pre></div>
<p>The data in the assay is a <code>DelayedMatrix</code> object:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">acc_b</span><span class="op">[[</span><span class="st">"RNASeq2GeneNorm"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="co">#&gt; [1] "DelayedMatrix"</span>
<span class="co">#&gt; attr(,"package")</span>
<span class="co">#&gt; [1] "DelayedArray"</span></code></pre></div>
<p>This is in contrast to the <code>acc</code> object that has matrices in memory:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">acc</span><span class="op">[[</span><span class="st">"RNASeq2GeneNorm"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>
<span class="co">#&gt;        TCGA-OR-A5J1-01A-11R-A29S-07 TCGA-OR-A5J2-01A-11R-A29S-07</span>
<span class="co">#&gt; DIRAS3                    1487.0317                       9.6631</span>
<span class="co">#&gt; MAPK14                     778.5783                    2823.6469</span>
<span class="co">#&gt; YAP1                      1009.6061                    2305.0590</span>
<span class="co">#&gt; CDKN1B                    2101.3449                    4248.9584</span>
<span class="co">#&gt; ERBB2                      651.2968                     246.4098</span>
<span class="co">#&gt;        TCGA-OR-A5J3-01A-11R-A29S-07</span>
<span class="co">#&gt; DIRAS3                      18.9602</span>
<span class="co">#&gt; MAPK14                    1061.7686</span>
<span class="co">#&gt; YAP1                      1561.2502</span>
<span class="co">#&gt; CDKN1B                    1348.5410</span>
<span class="co">#&gt; ERBB2                       90.0607</span>
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assays</a></span><span class="op">(</span><span class="va">acc</span><span class="op">[[</span><span class="st">"RNASeq2GeneNorm"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] "matrix" "array"</span></code></pre></div>
</div>
</div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<ul>
<li><p><a href="https://www.biorxiv.org/content/10.1101/2021.06.01.445670">Muon: multimodal omics analysis framework</a> preprint</p></li>
<li><p><a href="https://mudata.readthedocs.io/">mudata</a> (Python) documentation</p></li>
<li><p>muon <a href="https://muon.readthedocs.io/">documentation</a> and <a href="https://gtca.github.io/muon/">web page</a></p></li>
</ul>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.1.2 (2021-11-01)</span>
<span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">#&gt; Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0</span>
<span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span>
<span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span>
<span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span>
<span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">#&gt; [8] base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] rhdf5_2.39.2                MultiAssayExperiment_1.20.0</span>
<span class="co">#&gt;  [3] SummarizedExperiment_1.24.0 Biobase_2.54.0             </span>
<span class="co">#&gt;  [5] GenomicRanges_1.46.1        GenomeInfoDb_1.30.0        </span>
<span class="co">#&gt;  [7] IRanges_2.28.0              MatrixGenerics_1.6.0       </span>
<span class="co">#&gt;  [9] matrixStats_0.61.0          MuData_0.99.0              </span>
<span class="co">#&gt; [11] S4Vectors_0.32.3            BiocGenerics_0.40.0        </span>
<span class="co">#&gt; [13] Matrix_1.3-4                BiocStyle_2.22.0           </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] XVector_0.34.0              compiler_4.1.2             </span>
<span class="co">#&gt;  [3] BiocManager_1.30.16         jquerylib_0.1.4            </span>
<span class="co">#&gt;  [5] rhdf5filters_1.6.0          zlibbioc_1.40.0            </span>
<span class="co">#&gt;  [7] bitops_1.0-7                tools_4.1.2                </span>
<span class="co">#&gt;  [9] SingleCellExperiment_1.16.0 digest_0.6.28              </span>
<span class="co">#&gt; [11] evaluate_0.14               memoise_2.0.1              </span>
<span class="co">#&gt; [13] lattice_0.20-45             rlang_0.4.12               </span>
<span class="co">#&gt; [15] DelayedArray_0.20.0         yaml_2.2.1                 </span>
<span class="co">#&gt; [17] pkgdown_1.6.1               xfun_0.28                  </span>
<span class="co">#&gt; [19] fastmap_1.1.0               GenomeInfoDbData_1.2.7     </span>
<span class="co">#&gt; [21] stringr_1.4.0               knitr_1.36                 </span>
<span class="co">#&gt; [23] desc_1.4.0                  fs_1.5.0                   </span>
<span class="co">#&gt; [25] systemfonts_1.0.3           rprojroot_2.0.2            </span>
<span class="co">#&gt; [27] grid_4.1.2                  R6_2.5.1                   </span>
<span class="co">#&gt; [29] textshaping_0.3.6           HDF5Array_1.22.1           </span>
<span class="co">#&gt; [31] rmarkdown_2.11              bookdown_0.24              </span>
<span class="co">#&gt; [33] Rhdf5lib_1.16.0             magrittr_2.0.1             </span>
<span class="co">#&gt; [35] htmltools_0.5.2             ragg_1.2.0                 </span>
<span class="co">#&gt; [37] stringi_1.7.6               RCurl_1.98-1.5             </span>
<span class="co">#&gt; [39] cachem_1.0.6                crayon_1.4.2</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Danila Bredikhin, Ilia Kats.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
