<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MuData">
<title>Getting started with MuData for MultiAssayExperiment • MuData</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting started with MuData for MultiAssayExperiment">
<meta property="og:description" content="MuData">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MuData</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-2">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Blood-CITE-seq.html">CITE-seq data with CiteFuse and MuData</a>
    <a class="dropdown-item" href="../articles/Cord-Blood-CITE-seq.html">CITE-seq data with MultiAssayExperiment and MuData</a>
    <a class="dropdown-item" href="../articles/Getting-Started.html">Getting started with MuData for MultiAssayExperiment</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/PMBio/MuDataMAE/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">
<script src="Getting-Started_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Getting started with MuData for MultiAssayExperiment</h1>
                        <h4 data-toc-skip class="author">Danila Bredikhin</h4>
            <address class="author_afil">
      European Molecular Biology Laboratory, Heidelberg, Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:danila.bredikhin@embl.de" class="email">danila.bredikhin@embl.de</a>
      </address>
                              <h4 data-toc-skip class="author">Ilia Kats</h4>
            <address class="author_afil">
      German Cancer Research Center, Heidelberg, Germany<br><a class="author_email" href="mailto:#"></a><a href="mailto:i.kats@dkfz-heidelberg.de" class="email">i.kats@dkfz-heidelberg.de</a>
      </address>
                  
            <h4 data-toc-skip class="date">2021-12-13</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/PMBio/MuDataMAE/blob/HEAD/vignettes/Getting-Started.Rmd" class="external-link"><code>vignettes/Getting-Started.Rmd</code></a></small>
      <div class="d-none name"><code>Getting-Started.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Multimodal data format — <a href="https://mudata.readthedocs.io/" class="external-link">MuData</a> — <a href="https://www.biorxiv.org/content/10.1101/2021.06.01.445670v1" class="external-link">has been introduced</a> to address the need for cross-platform standard for sharing large-scale multimodal omics data. Importantly, it develops ideas of and is compatible with <a href="https://anndata.readthedocs.io/" class="external-link">AnnData</a> standard for storing raw and derived data for unimodal datasets.</p>
<p>In Bioconductor, multimodal datasets have been stored in <a href="https://bioconductor.org/packages/MultiAssayExperiment" class="external-link">MultiAssayExperiment</a> (MAE) objects. This <code>MuData</code> package provides functionality to read data from MuData files into MAE objects as well as to save MAE objects into H5MU files.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>Stable version of <code>MuData</code> will be available in future bioconductor versions.</p>
<p>BiocManager::install(“MuData”)</p>
<p>The most recent dev build can be installed from GitHub:</p>
<p>library(remotes)remotes::install_github(“PMBio/MuDataMAE”)</p>
</div>
<div class="section level2">
<h2 id="loading-libraries">Loading libraries<a class="anchor" aria-label="anchor" href="#loading-libraries"></a>
</h2>
<p>library(MuData)library(MultiAssayExperiment)library(rhdf5)</p>
</div>
<div class="section level2">
<h2 id="writing-h5mu-files">Writing H5MU files<a class="anchor" aria-label="anchor" href="#writing-h5mu-files"></a>
</h2>
<p>We’ll use a simple MAE object from the <code>MultiAssayExperiment</code> package that we’ll then save in a H5MU file.</p>
<p>data(miniACC)miniACC#&gt; A MultiAssayExperiment object of 5 listed #&gt; experiments with user-defined names and respective classes. #&gt; Containing an ExperimentList class object of length 5: #&gt; [1] RNASeq2GeneNorm: SummarizedExperiment with 198 rows and 79 columns #&gt; [2] gistict: SummarizedExperiment with 198 rows and 90 columns #&gt; [3] RPPAArray: SummarizedExperiment with 33 rows and 46 columns #&gt; [4] Mutations: matrix with 97 rows and 90 columns #&gt; [5] miRNASeqGene: SummarizedExperiment with 471 rows and 80 columns #&gt; Functionality: #&gt; experiments() - obtain the ExperimentList instance #&gt; colData() - the primary/phenotype DataFrame #&gt; sampleMap() - the sample coordination DataFrame #&gt; <code>$</code>, <code>[</code>, <code>[[</code> - extract colData columns, subset, or experiment #&gt; *Format() - convert into a long or wide DataFrame #&gt; assays() - convert ExperimentList to a SimpleList of matrices #&gt; exportClass() - save data to flat files</p>
<p>We will now write its contents into an H5MU file with <code>WriteH5MU</code>:</p>
<p>WriteH5MU(miniACC, “miniacc.h5mu”)</p>
</div>
<div class="section level2">
<h2 id="reading-h5mu-files">Reading H5MU files<a class="anchor" aria-label="anchor" href="#reading-h5mu-files"></a>
</h2>
<p>We can manually check the top level of the structure of the file:</p>
<p>rhdf5::h5ls(“miniacc.h5mu”, recursive = FALSE)#&gt; group name otype dclass dim #&gt; 0 / mod H5I_GROUP<br>
#&gt; 1 / obs H5I_GROUP<br>
#&gt; 2 / obsm H5I_GROUP<br>
#&gt; 3 / obsmap H5I_GROUP<br>
#&gt; 4 / var H5I_GROUP</p>
<p>Or dig deeper into the file:</p>
<p>h5 &lt;- rhdf5::H5Fopen(“miniacc.h5mu”)h5&amp;‘mod’#&gt; HDF5 GROUP #&gt; name /mod #&gt; filename #&gt; #&gt; name otype dclass dim #&gt; 0 Mutations H5I_GROUP<br>
#&gt; 1 RNASeq2GeneNorm H5I_GROUP<br>
#&gt; 2 RPPAArray H5I_GROUP<br>
#&gt; 3 gistict H5I_GROUP<br>
#&gt; 4 miRNASeqGene H5I_GROUP rhdf5::H5close()</p>
<div class="section level3">
<h3 id="creating-mae-objects-from-h5mu-files">Creating MAE objects from H5MU files<a class="anchor" aria-label="anchor" href="#creating-mae-objects-from-h5mu-files"></a>
</h3>
<p>This package provides <code>ReadH5MU</code> to create an object with data from an H5MU file. Since H5MU structure has been designed to accommodate more structured information than MAE, only some data will be read. For instance, MAE has no support for loading multimodal embeddings or pairwise graphs.</p>
<p>acc &lt;- ReadH5MU(“miniacc.h5mu”)acc#&gt; A MultiAssayExperiment object of 5 listed #&gt; experiments with user-defined names and respective classes. #&gt; Containing an ExperimentList class object of length 5: #&gt; [1] Mutations: SingleCellExperiment with 97 rows and 90 columns #&gt; [2] RNASeq2GeneNorm: SingleCellExperiment with 198 rows and 79 columns #&gt; [3] RPPAArray: SingleCellExperiment with 33 rows and 46 columns #&gt; [4] gistict: SingleCellExperiment with 198 rows and 90 columns #&gt; [5] miRNASeqGene: SingleCellExperiment with 471 rows and 80 columns #&gt; Functionality: #&gt; experiments() - obtain the ExperimentList instance #&gt; colData() - the primary/phenotype DataFrame #&gt; sampleMap() - the sample coordination DataFrame #&gt; <code>$</code>, <code>[</code>, <code>[[</code> - extract colData columns, subset, or experiment #&gt; *Format() - convert into a long or wide DataFrame #&gt; assays() - convert ExperimentList to a SimpleList of matrices #&gt; exportClass() - save data to flat files</p>
<p>Importantly, we recover the information from the original MAE object:</p>
<p>head(colData(miniACC)[,1:4])#&gt; DataFrame with 6 rows and 4 columns #&gt; patientID years_to_birth vital_status days_to_death #&gt; <character><integer><integer><integer> #&gt; TCGA-OR-A5J1 TCGA-OR-A5J1 58 1 1355 #&gt; TCGA-OR-A5J2 TCGA-OR-A5J2 44 1 1677 #&gt; TCGA-OR-A5J3 TCGA-OR-A5J3 23 0 NA #&gt; TCGA-OR-A5J4 TCGA-OR-A5J4 23 1 423 #&gt; TCGA-OR-A5J5 TCGA-OR-A5J5 30 1 365 #&gt; TCGA-OR-A5J6 TCGA-OR-A5J6 29 0 NA head(colData(acc)[,1:4])#&gt; DataFrame with 6 rows and 4 columns #&gt; patientID years_to_birth vital_status days_to_death #&gt; <character><integer><integer><integer> #&gt; TCGA-OR-A5J1 TCGA-OR-A5J1 58 1 1355 #&gt; TCGA-OR-A5J2 TCGA-OR-A5J2 44 1 1677 #&gt; TCGA-OR-A5J3 TCGA-OR-A5J3 23 0 NA #&gt; TCGA-OR-A5J4 TCGA-OR-A5J4 23 1 423 #&gt; TCGA-OR-A5J5 TCGA-OR-A5J5 30 1 365 #&gt; TCGA-OR-A5J6 TCGA-OR-A5J6 29 0 NA</integer></integer></integer></character></integer></integer></integer></character></p>
<p>Features metadata is also recovered:</p>
<p>head(rowData(miniACC[[“gistict”]]))#&gt; DataFrame with 6 rows and 3 columns #&gt; Gene.Symbol Locus.ID Cytoband #&gt; <character><character><character> #&gt; DIRAS3 DIRAS3 9077 1p31.3 #&gt; MAPK14 MAPK14 1432 6p21.31 #&gt; YAP1 YAP1 10413 11q22.1 #&gt; CDKN1B CDKN1B 1027 12p13.1 #&gt; ERBB2 ERBB2 2064 17q12 #&gt; G6PD G6PD 2539 Xq28 head(rowData(acc[[“gistict”]]))#&gt; DataFrame with 6 rows and 3 columns #&gt; Gene.Symbol Locus.ID Cytoband #&gt; <character><character><character> #&gt; DIRAS3 DIRAS3 9077 1p31.3 #&gt; MAPK14 MAPK14 1432 6p21.31 #&gt; YAP1 YAP1 10413 11q22.1 #&gt; CDKN1B CDKN1B 1027 12p13.1 #&gt; ERBB2 ERBB2 2064 17q12 #&gt; G6PD G6PD 2539 Xq28</character></character></character></character></character></character></p>
<div class="section level4">
<h4 id="backed-objects">Backed objects<a class="anchor" aria-label="anchor" href="#backed-objects"></a>
</h4>
<p>It is possible to read H5MU files while keeping matrices (both <code>.X</code> and <code>.layers</code>) on disk.</p>
<p>acc_b &lt;- ReadH5MU(“miniacc.h5mu”, backed = TRUE)assays(acc_b[[“RNASeq2GeneNorm”]])[[1]][1:5,1:3]#&gt; &lt;5 x 3&gt; matrix of class DelayedMatrix and type “double”: #&gt; TCGA-OR-A5J1-01A-11R-A29S-07 TCGA-OR-A5J2-01A-11R-A29S-07 #&gt; DIRAS3 1487.0317 9.6631 #&gt; MAPK14 778.5783 2823.6469 #&gt; YAP1 1009.6061 2305.0590 #&gt; CDKN1B 2101.3449 4248.9584 #&gt; ERBB2 651.2968 246.4098 #&gt; TCGA-OR-A5J3-01A-11R-A29S-07 #&gt; DIRAS3 18.9602 #&gt; MAPK14 1061.7686 #&gt; YAP1 1561.2502 #&gt; CDKN1B 1348.5410 #&gt; ERBB2 90.0607</p>
<p>The data in the assay is a <code>DelayedMatrix</code> object:</p>
<p>class(assays(acc_b[[“RNASeq2GeneNorm”]])[[1]])#&gt; [1] “DelayedMatrix” #&gt; attr(,“package”) #&gt; [1] “DelayedArray”</p>
<p>This is in contrast to the <code>acc</code> object that has matrices in memory:</p>
<p>assays(acc[[“RNASeq2GeneNorm”]])[[1]][1:5,1:3]#&gt; TCGA-OR-A5J1-01A-11R-A29S-07 TCGA-OR-A5J2-01A-11R-A29S-07 #&gt; DIRAS3 1487.0317 9.6631 #&gt; MAPK14 778.5783 2823.6469 #&gt; YAP1 1009.6061 2305.0590 #&gt; CDKN1B 2101.3449 4248.9584 #&gt; ERBB2 651.2968 246.4098 #&gt; TCGA-OR-A5J3-01A-11R-A29S-07 #&gt; DIRAS3 18.9602 #&gt; MAPK14 1061.7686 #&gt; YAP1 1561.2502 #&gt; CDKN1B 1348.5410 #&gt; ERBB2 90.0607 print(class(assays(acc[[“RNASeq2GeneNorm”]])[[1]]))#&gt; [1] “matrix” “array”</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li><p><a href="https://www.biorxiv.org/content/10.1101/2021.06.01.445670" class="external-link">Muon: multimodal omics analysis framework</a> preprint</p></li>
<li><p><a href="https://mudata.readthedocs.io/" class="external-link">mudata</a> (Python) documentation</p></li>
<li><p>muon <a href="https://muon.readthedocs.io/" class="external-link">documentation</a> and <a href="https://gtca.github.io/muon/" class="external-link">web page</a></p></li>
</ul>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>sessionInfo()#&gt; R Under development (unstable) (2021-12-11 r81350) #&gt; Platform: x86_64-pc-linux-gnu (64-bit) #&gt; Running under: Ubuntu 20.04.3 LTS #&gt; #&gt; Matrix products: default #&gt; BLAS: /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 #&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0 #&gt; #&gt; locale: #&gt; [1] LC_CTYPE=C.UTF-8 LC_NUMERIC=C LC_TIME=C.UTF-8<br>
#&gt; [4] LC_COLLATE=C.UTF-8 LC_MONETARY=C.UTF-8 LC_MESSAGES=C.UTF-8<br>
#&gt; [7] LC_PAPER=C.UTF-8 LC_NAME=C LC_ADDRESS=C<br>
#&gt; [10] LC_TELEPHONE=C LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C<br>
#&gt; #&gt; attached base packages: #&gt; [1] stats4 stats graphics grDevices utils datasets methods<br>
#&gt; [8] base<br>
#&gt; #&gt; other attached packages: #&gt; [1] rhdf5_2.39.2 MultiAssayExperiment_1.21.5 #&gt; [3] SummarizedExperiment_1.25.3 Biobase_2.55.0<br>
#&gt; [5] GenomicRanges_1.47.5 GenomeInfoDb_1.31.1<br>
#&gt; [7] IRanges_2.29.1 MatrixGenerics_1.7.0<br>
#&gt; [9] matrixStats_0.61.0 MuData_0.99.0<br>
#&gt; [11] S4Vectors_0.33.8 BiocGenerics_0.41.2<br>
#&gt; [13] Matrix_1.4-0 BiocStyle_2.23.0<br>
#&gt; #&gt; loaded via a namespace (and not attached): #&gt; [1] XVector_0.35.0 bslib_0.3.1<br>
#&gt; [3] compiler_4.2.0 BiocManager_1.30.16<br>
#&gt; [5] jquerylib_0.1.4 rhdf5filters_1.7.0<br>
#&gt; [7] zlibbioc_1.41.0 bitops_1.0-7<br>
#&gt; [9] tools_4.2.0 SingleCellExperiment_1.17.2 #&gt; [11] digest_0.6.29 jsonlite_1.7.2<br>
#&gt; [13] evaluate_0.14 memoise_2.0.1<br>
#&gt; [15] lattice_0.20-45 rlang_0.4.12<br>
#&gt; [17] DelayedArray_0.21.2 yaml_2.2.1<br>
#&gt; [19] pkgdown_2.0.1 xfun_0.28<br>
#&gt; [21] fastmap_1.1.0 GenomeInfoDbData_1.2.7<br>
#&gt; [23] stringr_1.4.0 knitr_1.36<br>
#&gt; [25] desc_1.4.0 fs_1.5.2<br>
#&gt; [27] sass_0.4.0 systemfonts_1.0.3<br>
#&gt; [29] rprojroot_2.0.2 grid_4.2.0<br>
#&gt; [31] R6_2.5.1 textshaping_0.3.6<br>
#&gt; [33] HDF5Array_1.23.2 rmarkdown_2.11<br>
#&gt; [35] bookdown_0.24 Rhdf5lib_1.17.0<br>
#&gt; [37] purrr_0.3.4 magrittr_2.0.1<br>
#&gt; [39] htmltools_0.5.2 ragg_1.2.1<br>
#&gt; [41] stringi_1.7.6 RCurl_1.98-1.5<br>
#&gt; [43] cachem_1.0.6 crayon_1.4.2</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Danila Bredikhin, Ilia Kats.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
